contracts/SecurityTokenRegistry.sol

```diff
@@ -666,8 +679,29 @@ contract SecurityTokenRegistry is EternalStorage, Proxy {
     function changeSecurityLaunchFee(uint256 _stLaunchFee) external onlyOwner {
         uint256 fee = getUintValue(STLAUNCHFEE);
         require(fee != _stLaunchFee, "Bad fee");
-        emit ChangeSecurityLaunchFee(fee, _stLaunchFee);
-        set(STLAUNCHFEE, _stLaunchFee);
+        _changeSecurityLaunchFee(fee, _stLaunchFee);
+    }
+
+    function _changeSecurityLaunchFee(uint256 _oldFee, uint256 _newFee) internal {
+        emit ChangeSecurityLaunchFee(_oldFee, _newFee);
+        set(STLAUNCHFEE, _newFee);
+    }
+
+    /**
+    * @notice Sets the ticker registration and ST launch fee amount and currency
+    * @param _tickerRegFee is the ticker registration fee (base 18 decimals)
+    * @param _stLaunchFee is the st generation fee (base 18 decimals)
+    * @param _isFeeInPoly defines if the fee is in poly or usd
+    */
+    function changeFeesAmountAndCurrency(uint256 _tickerRegFee, uint256 _stLaunchFee, bool _isFeeInPoly) external onlyOwner {
+        uint256 tickerFee = getUintValue(TICKERREGFEE);
+        uint256 stFee = getUintValue(STLAUNCHFEE);
+        bool isOldFeesInPoly = getBoolValue(IS_FEE_IN_POLY);
+        require(isOldFeesInPoly != _isFeeInPoly, "Currency unchanged");
```

Comment:
So you can only use this function if you're changing from USD <> POLY fees (not if you're just changing the fee amounts)?

Reply:
I added this check later as a sanity check. We already have functions to change amounts, If I remove this check, there is no sanity check left and tx will go through even if the owner forgot to enter any data and calls the function with 000000...000.

If the fee is already in poly then the owner will still be allowed to enter 000000.000 so this is not a foolproof check and I am rather agnostic about keeping it. Let me know if you think we should remove the check to allow the owner to change both prices in a single transaction.